src/OSPSuite.Core/Domain/Descriptors/DescriptorCriteria.cs

```diff
@@ -12,14 +12,14 @@ public override string ToString()
          return this.ToString(Constants.AND.ToUpper(), " ").Trim(' ');
       }
 
-      public bool IsSatisfiedBy(Tags tags)
+      public bool IsSatisfiedBy(EntityDescriptor entityDescriptor)
       {
-         return this.Any() && this.All(condition => condition.IsSatisfiedBy(tags));
+         return this.Any() && this.All(condition => condition.IsSatisfiedBy(entityDescriptor));
       }
 
       public virtual bool IsSatisfiedBy(IEntity entity)
       {
-         return IsSatisfiedBy(new Tags(entity.Tags) {new Tag(entity.Name)});
+         return IsSatisfiedBy(new EntityDescriptor(entity));
```

Comment:
Not all tags are the same. there is now a hiearchy of tags and you cannot just check everything